Replace the column resizing process with one based on drag & drop events [PREVIEW]

nicoe opened the pull request:
Fix PCLAS-1066

JCavallo committed:
gtk: [6.8-patches] Force py38 compatibility [CUSTOM] (#179)

JCavallo committed:
Fix iteration on rpc_entries [PREVIEW]

JCavallo committed:
Revert "sao: [6.8-patches] Fix iteration on rpc_entries [CUSTOM]"

JCavallo committed:
sao: Unlink parent of O2M when setting defaults [CUSTOM]

This is hacky, but allows to have the same behaviour than
that of gtk / proteus

Fix PCLAS-713

JCavallo committed:
sao: [6.8-patches] Add slight color on selected list-form [CUSTOM]

JCavallo committed:
sao: [6.8-patches] Disable yexpand for now [CUSTOM]

JCavallo committed:
sao: [6.8-patches] Make selections without any values readonly [CUSTOM] (#180)

Fix PCLAS-734

nicoe committed:
Select first record in tree views [CUSTOM] (#181)

Fix PCLAS-726

JCavallo committed:
sao: [6.8-patches] Do not expand invisible widgets (#184)

Fix PCLAS-753

nicoe committed:
Add ellipsis on column header [CUSTOM] (#185)

Fix PCLAS-597

Co-authored-by: Jean Cavallo <jean.cavallo@hotmail.fr>

nicoe committed:
Update column visibility at the end of rendering the treeview [PREVIEW] (#187)

For performance reasons it's better not to mix DOM reading and
modifying especially in JS loops.

PCLAS-772

nicoe committed:
Select first record of treeview if none is selected [CUSTOM] (#189)

PCLAS-767

nicoe committed:
Use a TableHandler method to fetch the estimated count of records [PREVIEW] (#191)

https://bugs.tryton.org/13118
PCLAS-784

nicoe committed:
Execute user login's actions sequentially [PREVIEW] (#190)

https://bugs.tryton.org/13074
Fix PCLAS-614

JCavallo committed:
Iteration on fbf91cdc0f: handle table queries

nicoe committed:
Correctly use COUNT [CUSTOM] (#193)

nicoe committed:
Transfer configuration, profiles and plugins from previous version [PREVIEW] (#50)

Co-authored-by: Jean Cavallo <jean.cavallo@hotmail.fr>

nicoe committed:
Add close button to "Filters" popup [CUSTOM] (#205)

Fix #CCOOG-642

nicoe committed:
Do not set overflow hidden on first column [PREVIEW] (#204)

Fix PCLAS-824

nicoe committed:
Force the use of 'en' when parsing XML files [CUSTOM] (#201)

https://bugs.tryton.org/13204
Fix PCLAS-816

BeatriceSchuster committed:
tryton: Change the translation of the 'show plain text' checkbox message [CUSTOM] (#188)

Fix #PJAZZ-2008

JCavallo committed:
sao: [6.8-patches] Fix layout of tab wizards (#208)

Fix PCLAS-843

nicoe committed:
Fetch name of wizards without a model [PREVIEW] (#207)

Fix PCLAS-770

nicoe committed:
Optimize instance creation in __getattr__ [PREVIEW] (#210)

* Optimize instance creation in __getattr__ [PREVIEW]

Accessing a M2O field from a X2M field could result in as many
instantiations as there are records in the X2M field. It could be a
performance bottleneck due to the many creations of Model kwargs and
potential useless calls of transaction.set_context().

Thus the optimization is twofold:

 - Create an additional code path when the M2O field modifies the
   context
 - Cache the kwargs creation process to prevent extra calls resulting in
   the same dictionary

* Use read to fetch reverse field value of O2M field in get [PREVIEW]

Using a browse there in order to get the id is a performance bottleneck
as it will result in the instanciation of potentially thousands of
models.

---------

Co-authored-by: Jean Cavallo <jean.cavallo@hotmail.fr>

nicoe committed:
Allow to resize attachment preview panel [PREVIEW] (#209)

https://bugs.tryton.org/13234
Fixes PCLAS-821

JCavallo committed:
trytond: [6.8-patches] Pass ids to read [PREVIEW] (#213)

Ref PCLAS-838

nicoe committed:
Allow VARCHAR / JSONB migration on PostgreSQL [CUSTOM] (#203)

Closes PCLAS-818

---------

Co-authored-by: Jean Cavallo <jean.cavallo@hotmail.fr>

JCavallo committed:
trytond: [6.8-patches] Revert getattr optimization

Unfix PCLAS-838

nicoe committed:
Skip translations refering to unknown XML ID or unactivated source (#214)

Fix PCLAS-572

nicoe committed:
Correctly compute column visibility for buttons (#212)

Fix PCLAS-828

JCavallo committed:
Revert "Skip translations refering to unknown XML ID or unactivated source (#214)"

This reverts commit 4667eea7aa7de44e266618d9815d6c0c414b9c87.

nicoe committed:
Optimize instance creation in ModelStorage.__getattr__ [PREVIEW]  (#219)

* Optimize instance creation in ModelStorage.__getattr__ [PREVIEW]

https://bugs.tryton.org/13245

* Sort X2M fields according to the order [PREVIEW]

Fix PCLAS-889
https://bugs.tryton.org/13288

JCavallo committed:
Correctly override _convert_domain [CUSTOM] (#221)

Fix PCLAS-901

nicoe committed:
Alert when using underspecified relation in domain [PREVIEW] (#217)

Fix PCLAS-874

---------

Co-authored-by: Jean Cavallo <jean.cavallo@hotmail.fr>

nicoe committed:
Correctly compute column visibility [CUSTOM] (#226)

nicoe committed:
Add a No-op context manager [CUSTOM] (#225)

Fix PCLAS-854

JCavallo committed:
debug: [6.8-patches] Improve views, fix generation

Fix PCLAS-946

nicoe committed:
Do not reset SQL indexes on each __setup__ call [PREVIEW] (#227)

https://bugs.tryton.org/13299
Fix PCLAS-880

JCavallo committed:
sao: [6.8-patches] Hide nested checkbox for selecting lines in tree (#232)

Fix PCLAS-964

JCavallo committed:
sao: [6.8-patches] Build using npm exec [CUSTOM]

JCavallo committed:
Skip translations refering to unknown XML ID or unactivated source [PREVIEW] (#215)

Fix PCLAS-883

---------

Co-authored-by: Nicolas Évrard <nicoe@openhex.org>

JCavallo committed:
Disable not null constraint drop for sqlite [CUSTOM] (#236)

Fix PCLAS-982

JCavallo committed:
gtk: [6.8-patches] Make text fields visually readonly (#241)

Fix PCLAS-1003

havaguernicoopengo committed:
sao: [6.8-patches] Fix border visibility in text forms [CUSTOM] (#240)

Fix PCLAS-993

---------

Co-authored-by: Jean Cavallo <jean.cavallo@hotmail.fr>

BeatriceSchuster committed:
tryton: [6.8-patches] Change the translation for the message shown when a required field is empty [CUSTOM] (#237)

Fix #PJAZZ-2244

JCavallo committed:
trytond: [6.8-patches] Fix translation res_id domain [PREVIEW] (#242)

Fix PCLAS-1009

nicoe committed:
Skip id field when setting default values of failed read calls [PREVIEW] (#247)

* Undo "Consider that records in exception are loaded [PREVIEW]"

* Skip id field when setting default values of failed read calls [PREVIEW]

Fix PCLAS-1006

nicoe committed:
Add missing Widget methods on source editor [CUSTOM] (#246)

Fix PCLAS-1005

nicoe committed:
Do not resize horizontally textarea [CUSTOM] (#245)

Fix PCLAS-996

nicoe committed:
Consider as local ordering expression using simple SQL statements [PREVIEW] (#248)

Rely on the tables dict to choose when to apply the OR-to-UNION optimization [PREVIEW]

Fix PCLAS-1015

Co-authored-by: Jean Cavallo <jean.cavallo@hotmail.fr>

JCavallo committed:
trytond: [6.8-patches] Do not crash on searches on None domains (#250)

Fix PCLAS-1047

JCavallo committed:
trytond: [6.8-patches] Do not over-optimize function active field searches (#251)

Fix PCLAS-1053

nicoe committed:
Correctly handle different return values from buttons [CUSTOM] (#243)

Fix PCLAS-990

nicoe committed:
Reset caches directly after sending the notification [CUSTOM] (#235)

Co-authored-by: Jean Cavallo <jean.cavallo@hotmail.fr>

JCavallo committed:
sao: [6.8-patches] Better usage of vertical space [CUSTOM] (#252)

Fix PCLAS-1055

havaguernicoopengo committed:
sao: [6.8-patches] open form in modal when already in modal (#253)

Fix PCLAS-1042

nicoe committed:
Resize columns in sao [PREVIEW] (#164)

Fix PCLAS-786

---------

Co-authored-by: Jean Cavallo <jean.cavallo@hotmail.fr>

nicoe committed:
Removal of account.payment.process.start view

nicoe committed:
The TimeoutException has been removed upstream

nicoe committed:
Correctly call status.start

nicoe committed:
Remove view_attributes method moved to coog_core

nicoe committed:
Add missing cursor and table definition from migration

nicoe committed:
Allows dict as value of x2Many [CUSTOM]

nicoe committed:
Use get_modules to get the module list [CUSTOM]

nicoe committed:
Create a stub Options object as it's used by custom migration code [CUSTOM]

nicoe committed:
The code needed to clean the pool is not needed in the new design [CUSTOM]

nicoe committed:
Add expected failures due to coopengo changes [CUSTOM]

nicoe committed:
Add missing with_transaction decorator

nicoe committed:
Allow to deepcopy Immutabledict [PREVIEW]

nicoe committed:
Temporarily deactivate the assert on _change_values [CUSTOM]

nicoe committed:
Fix typo

nicoe committed:
Correctly use test_name when testing access [PREVIEW]

nicoe committed:
Use upstream API name to get timezone [PREVIEW]

JCavallo committed:
trytond: [6.8-patches] Add hook to control log creation [CUSTOM] (#259)

Fix PCLAS-1082

JCavallo committed:
sao: [6.8-patches] Better column resizing (#257)

JCavallo committed:
trytond: clear database _extensions between tests [TO REVIEW]

JCavallo committed:
trytond: remove duplicate 'indexes' parameter [TO REVIEW]

JCavallo committed:
coopengo_modules: Migrate ir.model.button->model [TO REVIEW]

JCavallo committed:
Use None for default value of view->module [TO REVIEW]

PierreCookie committed:
report: Fix custom reporting (#269) [CUSTOM]

Fix #PROCK-2279

JCavallo committed:
modules: Use dummy validation method for scenarios [CUSTOM]

JCavallo committed:
trytond: Create access right for warnings [TO REVIEW]

JCavallo committed:
debug: Fix possible domain fetch

JCavallo committed:
[account_payment_clearing]: Remove uneeded process step in test

JCavallo committed:
[account_statement]: Adapt tests to dummy_validate_method

JCavallo committed:
[analytic_invoice]: Disable invoice_defer test scenario [CUSTOM]

JCavallo committed:
account_statement: Update test scenario to use dummy_validate_method [CUSTOM]

JCavallo committed:
account_statement: Update test scenario to use dummy_validate_method [CUSTOM]

JCavallo committed:
sao: select_row may be called without an event [PREVIEW] (#272)

Fix PCLAS-1120

JCavallo committed:
sao: Fix typo following merge

JCavallo committed:
tryton: Add missing import

JCavallo committed:
Add missing import for canonicalize

JCavallo committed:
trytond: Update TimeoutException to new version

JCavallo committed:
tryton: Use _get_icon rather than register_icon [CUSTOM]

PierreCookie committed:
trytond: Improve indexes cleaning with trimmed named (#270) [CUSTOM]

Fix PROCK-1931

nicoe committed:
Allow to hide search bar [CUSTOM] (#267)

Fix PCLAS-1060

nicoe committed:
Add expandable action on x2m title [CUSTOM] (#268)

Fix PCLAS-1061

nicoe committed:
Use right path for grunt watch [CUSTOM]

nicoe committed:
Fix starting wizard size issues [CUSTOM]

nicoe committed:
Make tab's title more visible [CUSTOM]

nicoe committed:
Fix main menu items visibility [CUSTOM]

nicoe committed:
Correctly import canonicalize [CUSTOM]

JCavallo committed:
Revert "trytond: Create access right for warnings [TO REVIEW]"

This reverts commit 57d4d3a2eb4ea55049f47a17177d0454bfdf1e42.

nicoe committed:
Fetch missing key definition in Record.set_on_change [PREVIEW] (#274)

Fix PCLAS-1113

JCavallo committed:
sao: Fix theme following merge 7.2 [CUSTOM]

nicoe committed:
Create x2m title widget in all cases [CUSTOM] (#278)

Fix PCLAS-1061

JCavallo committed:
sao: Remove extra padding in act windows [CUSTOM]

JCavallo committed:
trytond: Remove index dumping mechanism [CUSTOM]

JCavallo committed:
debug: Update init to use new pool API [CUSTOM]

JCavallo committed:
sao: Vertical align XXX2Many toolbar content (#277)

Issue link: [https://coopengo.atlassian.net/browse/PCLAS-1134](https://coopengo.atlassian.net/browse/PCLAS-1134)

Fix PCLAS-1134

nicoe committed:
Do not set instance buttons inactive in a O2M [PREVIEW] (#279)

Fix PCLAS-1124

Commit:
Packaging coog-24.38.2438-rc3

JCavallo committed:
account_statement: Add amount_second_currency in line group table query [PREVIEW]

JCavallo committed:
Revert "account_statement: Add amount_second_currency in line group table query [PREVIEW]"

This reverts commit bddcbbe3d0dfb9d44f4f603a8485c7d7293a1563.

nicoe committed:
Handle handling_sum attribut when parsing field definition [CUSTOM] (#281)

Fix PCLAS-1149

nicoe committed:
Correctly get display counters of cell caches [CUSTOM] (#280)

Fix PCLAS-1148

nicoe committed:
Add refresh button on context screens [CUSTOM] (#266)

Fix PCLAS-1059

JCavallo committed:
tryton: Remove leftover of multi mixed view [CUSTOM] (#283)

Issue link: [https://coopengo.atlassian.net/browse/PCLAS-1156](https://coopengo.atlassian.net/browse/PCLAS-1156)

Fix PCLAS-1156

JCavallo committed:
trytond: Fix PYSON repr of Eval using Pyson as default (#282)

Issue link: [https://coopengo.atlassian.net/browse/PCLAS-1151](https://coopengo.atlassian.net/browse/PCLAS-1151)

Fix PCLAS-1151

nicoe committed:
Do not store button column width [PREVIEW] (#284)

Fix PCLAS-1157

JCavallo committed:
sao: Update build file for 7.2

Commit:
Packaging coog-24.39.2439-rc1

JCavallo committed:
sao: Update build script [CUSTOM]

JCavallo committed:
sao: Remove scrollbar below calendar icon (#286)

Issue link: [https://coopengo.atlassian.net/browse/PCLAS-1165](https://coopengo.atlassian.net/browse/PCLAS-1165)

Fix PCLAS-1165

JCavallo committed:
sao: Vertical align labels with field values (#287)

Issue link: [https://coopengo.atlassian.net/browse/PCLAS-1167](https://coopengo.atlassian.net/browse/PCLAS-1167)

Fix PCLAS-1167

JCavallo committed:
Revert "Fetch missing key definition in Record.set_on_change [PREVIEW] (#274)"

This reverts commit 494a6d39918b9d54b06345cd529912b84d3c17e6.

xungongcoopengo committed:
party: Add a hook before the application of a party replacement #218 (#289)

Fix PROCK-1699

nicoe committed:
Return Screen's display promise when creating a new one [PREVIEW] (#291)

* Revert "Fetch missing key definition in Record.set_on_change [PREVIEW] (#274)"

This reverts commit 494a6d39918b9d54b06345cd529912b84d3c17e6.

* Return Screen's display promise when creating a new one [PREVIEW]

Returning this promise ensure the correct scheduling of the promises
when a wizard contains an x2M widgets that will create a new Sao.Model
which will receive later its fields definition while the on_change will
trigger the process to set the record's values.

---------

Co-authored-by: Jean Cavallo <jean.cavallo@hotmail.fr>

nicoe committed:
Display filter by default on screen [CUSTOM] (#296)

* Display filter by default on screen [CUSTOM]

Fix PCLAS-1179

* Fix sao as well

---------

Co-authored-by: Jean Cavallo <jean.cavallo@hotmail.fr>

nicoe committed:
Use the right index of the resized column [CUSTOM] (#295)

Fix PCLAS-1158

Commit:
Packaging coog-24.40.2440-rc1

JCavallo committed:
sao: Style tabs to be more visible (#298)

Issue link: [https://coopengo.atlassian.net/browse/PCLAS-1198](https://coopengo.atlassian.net/browse/PCLAS-1198)

Fix PCLAS-1198

JCavallo committed:
sao: Remove dropdown arrow on readonly selections (#300)

Issue link: [https://coopengo.atlassian.net/browse/PCLAS-1210](https://coopengo.atlassian.net/browse/PCLAS-1210)

Fix PCLAS-1210

JCavallo committed:
sao: Fix resizing of last column (#301)

Issue link: [https://coopengo.atlassian.net/browse/PCLAS-1212](https://coopengo.atlassian.net/browse/PCLAS-1212)

Fix PCLAS-1212

JCavallo committed:
sao: Always use the same size for numeric widgets (#302)

Issue link: [https://coopengo.atlassian.net/browse/PCLAS-1214](https://coopengo.atlassian.net/browse/PCLAS-1214)

Fix PCLAS-1214

JCavallo committed:
sao: Fix vertical size of textareas (#303)

Issue link: [https://coopengo.atlassian.net/browse/PCLAS-1216](https://coopengo.atlassian.net/browse/PCLAS-1216)

Fix PCLAS-1216

nicoe committed:
Return the new record in promises from Screen.new_ [PREVIEW] (#299)

Fix PCLAS-1201

Co-authored-by: Jean Cavallo <jean.cavallo@hotmail.fr>

JCavallo committed:
trytond: Do not call getter without any records to get [PREVIEW] (#304)

Issue link: [https://coopengo.atlassian.net/browse/PCLAS-1220](https://coopengo.atlassian.net/browse/PCLAS-1220)

Fix PCLAS-1220

nicoe committed:
Revert "sao / gtk: Do not trigger an extra read for an empty rec_name [CUSTOM] (#170)" (#297)

This reverts commit ce201f1454f7604a5b8d4b3b7de842a1ec22febf.

nicoe committed:
Update only the actual ir.model and ir.model.fields translations [PREVIEW] (#294)

Fix PCLAS-1014
https://bugs.tryton.org/13546

nicoe committed:
Do not request a size for collapsed x2m widgets [CUSTOM] (#307)

Fix PCLAS-1202

Commit:
Packaging coog-24.41.2441-rc3

AdrienBenduc committed:
tryton: refresh translation (#316)

Fix #PMETA-1957

nicoe committed:
Raise the original RPCException when handling it in RPCProgress [PREVIEW] (#310)

Fix PCLAS-1224

nicoe committed:
Do not include inexistant ssl dir and use right Executable param name [CUSTOM] (#292)

nicoe committed:
Remove merge markers from stock module [CUSTOM] (#315)

Fix PCLAS-XXXX

nicoe committed:
Allow searching on subdivision shorter code when searching on a pattern [PREVIEW] (#308)

Fix PCLAS-1196

JCavallo committed:
coopengo_modules: Update copyright end date [CUSTOM]

Commit:
Packaging coog-24.42.2442-rc1

JCavallo committed:
sao: Reduce padding between cards widgets (#313)

Issue link: [https://coopengo.atlassian.net/browse/PCLAS-1246](https://coopengo.atlassian.net/browse/PCLAS-1246)

Fix PCLAS-1246

JCavallo committed:
sao: Remove padding for XXX2Many all alone in a page [CUSTOM] (#314)

Issue link: [https://coopengo.atlassian.net/browse/PCLAS-1250](https://coopengo.atlassian.net/browse/PCLAS-1250)

Fix PCLAS-1250

nicoe committed:
Reloading context updates the user information [CUSTOM] (#285)

Fix PCLAS-852

Co-authored-by: Jean Cavallo <jean.cavallo@hotmail.fr>

JCavallo committed:
sao: Block interactions if waiting more than 200ms [CUSTOM] (#320)

Issue link: [https://coopengo.atlassian.net/browse/PCLAS-1267](https://coopengo.atlassian.net/browse/PCLAS-1267)

Fix PCLAS-1267

BeatriceSchuster committed:
trytond: Fix test_view method to skip checks on 'refresh parent' buttons [CUSTOM] (#309)

Fix #PJAZZ-2467

BeatriceSchuster committed:
coopengo_modules: Move the modules "electronic_signature" and "cryptolog" to coog [CUSTOM] (#275)

Fix #PJAZZ-2183

JCavallo committed:
sao: Adapt textarea height to contents [CUSTOM] (#319)

Issue link: [https://coopengo.atlassian.net/browse/PCLAS-1264](https://coopengo.atlassian.net/browse/PCLAS-1264)

Fix PCLAS-1264

nicoe committed:
Close the tab related to the screen calling the 'close' client action [CUSTOM] (#305)

* Close the tab related to the screen calling the 'close' client action [PREVIEW]

https://bugs.tryton.org/13559
https://bugs.tryton.org/13560

Fix PCLAS-1182

* Add debugging tool [CUSTOM]

* Return a promise from widgets display method [CUSTOM]

This allows for the scheduling of actions after the display of a view
has been finalized.

The display methods of the x2M widgets must take care of the fact that
the screen display being called afterwards should be return instead of
the maybe already resolve promise from the screen initial switch view.

* Resolve the wizard promise only when the destroy action is terminated

nicoe committed:
Remove handling of the system date from tryton [CUSTOM] (#306)

Fix PCLAS-852

JCavallo committed:
trytond: Reraise original error if no handler [CUSTOM] (#326)

Issue link: [https://coopengo.atlassian.net/browse/PCLAS-1287](https://coopengo.atlassian.net/browse/PCLAS-1287)

Fix PCLAS-1287

Commit:
Packaging coog-24.43.2442

JCavallo committed:
sao: Highlight booleans in editable tree views [CUSTOM] (#329)

Issue link: [https://coopengo.atlassian.net/browse/PCLAS-1291](https://coopengo.atlassian.net/browse/PCLAS-1291)

Fix PCLAS-1291

JCavallo committed:
sao: Allow to vertically resize lists [CUSTOM] (#322)

Issue link: [https://coopengo.atlassian.net/browse/PCLAS-1274](https://coopengo.atlassian.net/browse/PCLAS-1274)

Fix PCLAS-1274

JCavallo committed:
trytond: Do not crash on avatar cache saving [CUSTOM] (#331)

Issue link: [https://coopengo.atlassian.net/browse/PCLAS-1302](https://coopengo.atlassian.net/browse/PCLAS-1302)

Fix PCLAS-1302

JCavallo committed:
sao: Better linting when pasting in CodeMirror [CUSTOM] (#332)

Issue link: [https://coopengo.atlassian.net/browse/PCLAS-1305](https://coopengo.atlassian.net/browse/PCLAS-1305)

Fix PCLAS-1305

nicoe committed:
Parse child node of group tags at the end [PREVIEW] (#330)

Fix PCLAS-1294

nicoe committed:
Add the missing gettext call to the reset password message [PREVIEW] (#328)

Fix PCLAS-1200

nicoe committed:
Simplify menu handling by using flex instead of offcanvas and make it resizeable [PREVIEW] (#327)

Fix PCLAS-595

nicoe committed:
Compute multi mixed widget form visibility after setting the current record from a new group [CUSTOM] (#333)

Fix PCLAS-1303

Co-authored-by: Jean Cavallo <jean.cavallo@hotmail.fr>

nicoe committed:
Use absolute positioning for the expander span [CUSTOM] (#324)

* Use absolute positioning for the expander span [CUSTOM]

Fix PCLAS-1168

This is required because the column resizing sets a width on the col and
the div inside the th which make the span tag stack.

We can use absolute positioning inside the th because they're sticky and
thus absolute will remove the span from the flow.

* Iteration for style

---------

Co-authored-by: Jean Cavallo <jean.cavallo@hotmail.fr>

JCavallo committed:
Revert "Compute multi mixed widget form visibility after setting the current record from a new group [CUSTOM] (#333)"

This reverts commit 5737ffd2ffea10093a214152943ad5d425d488b4.

JCavallo committed:
Revert "Parse child node of group tags at the end [PREVIEW] (#330)"

This reverts commit 3c4d2a9059797f87a9639caebbb8bf4e60a7cc12.

Commit:
Packaging coog-24.44.2444-rc2

JCavallo committed:
sao: Return empty promise in display [PREVIEW?] (#334)

Issue link: [https://coopengo.atlassian.net/browse/PCLAS-1312](https://coopengo.atlassian.net/browse/PCLAS-1312)

Fix PCLAS-1312

Commit:
Packaging coog-24.45.2445-rc2

nicoe committed:
Find new key names using the definition from the widget's field [PREVIEW] (#335)

* Reapply "Parse child node of group tags at the end [PREVIEW] (#330)"

This reverts commit b3948934d71fb336fd3746f35e85186ac6e45c6f.

* Reapply "Compute multi mixed widget form visibility after setting the current record from a new group [CUSTOM] (#333)"

This reverts commit 2261fe89b29003837248fc0f78d4822fb9ca659e.

* Find new key names using the definition from the widget's field [PREVIEW]

Fix PCLAS-1308

JCavallo committed:
sao: Save current tab on Ctrl+s in rule engine [CUSTOM] (#339)

Issue link: [https://coopengo.atlassian.net/browse/PCLAS-1323](https://coopengo.atlassian.net/browse/PCLAS-1323)

Fix PCLAS-1323

nicoe committed:
Do not take into account twice the always present menu column [CUSTOM] (#337)

* Do not take into account twice the always present menu column [CUSTOM]

Fix PCLAS-1318

* Remove resized cache, handle last invisible columns

* Typo

---------

Co-authored-by: Jean Cavallo <jean.cavallo@hotmail.fr>

nicoe committed:
Apply rounding when setting values on Float and Numeric fields [PREVIEW] (#336)

Fix PCLAS-1300

nicoe committed:
Use a flexbox display for wizards in forms [CUSTOM] (#325)

Fix PCLAS-1258

nicoe committed:
Fill the source value of fmany2one in create and write methods [PREVIEW] (#341)

https://bugs.tryton.org/13522
PCLAS-1336

Commit:
Packaging coog-24.46.2446-rc1

nicoe committed:
Use EXISTS instead of IN for M2O's where [CUSTOM] (#342)

Fix PCLAS-1338

JCavallo committed:
Revert "Use EXISTS instead of IN for M2O's where [CUSTOM] (#342)" (#344)

This reverts commit 48f30f3279e7c2717fb7e4efe535b296a8436ddb.

nicoe committed:
Use EXISTS when using a 'where' operator in Many2One [PREVIEW] (#345)

https://bugs.tryton.org/11935
Fix PCLAS-1338

nicoe committed:
Delete warning at the end of the transaction (#348)

Fix PCLAS-1335

nicoe committed:
Activate the horizontal scroll on tabcontent [PREVIEW] (#338)

* Activate the horizontal scroll on tabcontent [CUSTOM]

Fix PCLAS-1320

* Revert "Activate the horizontal scroll on tabcontent [CUSTOM]"

This reverts commit 2edae2802cbac45e56b6dafa2dee206e2e4305e7.

* Overflow horizontally the tab content when necessary [PREVIEW]

https://bugs.tryton.org/13657

* Handle scrollbar in tabbed wizards

* Use upstream better version

* Styling

* Allow scrolling on the body of the tabbed wizards

---------

Co-authored-by: Jean Cavallo <jean.cavallo@hotmail.fr>

nicoe committed:
Add context menu on Many2One [CUSTOM] (#346)

* Add context menu on Many2One [CUSTOM]

Fix PCLAS-1354

* Update French translation

nicoe committed:
Do not check_access when deleting Warnings [PREVIEW] (#349)

Fix PCLAS-1335

JCavallo committed:
sao: Do not reset view on algorithm focus out [CUSTOM] (#351)

Issue link: [https://coopengo.atlassian.net/browse/PCLAS-1375](https://coopengo.atlassian.net/browse/PCLAS-1375)

Fix PCLAS-1375

Commit:
Packaging coog-24.47.2447

Commit:
Packaging coog-24.48.2448-rc1

JCavallo committed:
trytond: Fix translated search on model name [PREVIEW] (#352)

Issue link: [https://coopengo.atlassian.net/browse/PCLAS-1381](https://coopengo.atlassian.net/browse/PCLAS-1381)

Fix PCLAS-1381

JCavallo committed:
trytond: Restore full debug log [CUSTOM] (#353)

Issue link: [https://coopengo.atlassian.net/browse/PCLAS-1387](https://coopengo.atlassian.net/browse/PCLAS-1387)

Fix PCLAS-1387

mamoun-rais committed:
party: Remove reports "Labels" and "Letter" on party [CUSTOM] (#318)

fix #pjazz-2434

mamoun-rais committed:
party: Fix after PJAZZ-2434 [CUSTOM] (#360)

* remove test_party_label_report

* comment test_party_label_report

nicoe committed:
Allow to resize the tree of available functions in the code editor [CUSTOM] (#357)


---------

Co-authored-by: Jean Cavallo <jean.cavallo@hotmail.fr>

nicoe committed:
Do not set the Group's context when setting a default group [PREVIEW] (#363)

https://bugs.tryton.org/13726
Fix PCLAS-1383

nicoe committed:
Ensure the whole M2M widget is displayed when the dropdown is shown [PREVIEW] (#356)

Fix PCLAS-1389

nicoe committed:
Call record modified when adding a record without a parent is a Group [PREVIEW] (#359)

https://bugs.tryton.org/13709
Fix PCLAS-1367

nicoe committed:
Allow trytond-admin set password of any user. Activate user if inactive. [PREVIEW] (#365)

https://bugs.tryton.org/13630
Fix CCOOG-1137

nicoe committed:
Set the overflow to visible only vertically [PREVIEW] (#366)

Fix PCLAS-1169

Commit:
Packaging coog-24.49.2449

nicoe committed:
Force ZIP64 compatibility for all reports [CUSTOM] (#323)

Fix PCLAS-1129

nicoe committed:
Replace the column resizing process with one based on drag & drop events [PREVIEW]

Fix PCLAS-1066

nicoe committed:
Reviewed the drag & drop implementation and fix CSS

Commit:
Packaging coog-24.50.2450

JCavallo committed:
trytond: Honor 'concurrently' when dropping indexes [PREVIEW] (#369)

Issue link: [https://coopengo.atlassian.net/browse/PCLAS-1415](https://coopengo.atlassian.net/browse/PCLAS-1415)

Fix PCLAS-1415

JCavallo committed:
country: Use iso3166-1 when importing translations (#370)

https://foss.heptapod.net/tryton/tryton/-/commit/15441b7fe3a26c070f128183a8cd9b95672d6438
Issue link: [https://coopengo.atlassian.net/browse/PCLAS-1432](https://coopengo.atlassian.net/browse/PCLAS-1432)

Fix PCLAS-1432

Commit:
Packaging coog-24.51.2451

nicoe committed:
Sync with upstream MR

Commit:
Packaging coog-24.52.2452

Commit:
Packaging coog-25.01.2501

JCavallo committed:
sao: Add debug tools to Sao (#376)

Issue link: [https://coopengo.atlassian.net/browse/PCLAS-1456](https://coopengo.atlassian.net/browse/PCLAS-1456)

Fix PCLAS-1456

JCavallo committed:
trytond: Add final init hooks [CUSTOM] (#374)

Issue link: [https://coopengo.atlassian.net/browse/PCLAS-1449](https://coopengo.atlassian.net/browse/PCLAS-1449)

Fix PCLAS-1449

Commit:
Packaging coog-25.02.2502-rc3

JCavallo committed:
sao: Safer developer tools

Ref PCLAS-1456

havaguernicoopengo committed:
debug: add previous button to debug (#377)

Fix PCLAS-1448

JCavallo committed:
trytond: Manually add language in _get_names_cache keys (#378)

Issue link: [https://coopengo.atlassian.net/browse/PCLAS-1468](https://coopengo.atlassian.net/browse/PCLAS-1468)

Fix PCLAS-1468

JCavallo committed:
sao: Fix type detection in Pyson [CUSTOM] (#379)

Issue link: [https://coopengo.atlassian.net/browse/PCLAS-1473](https://coopengo.atlassian.net/browse/PCLAS-1473)

Fix PCLAS-1473

nicoe committed:
Set selection size according to its content [CUSTOM] (#367)

* Set selection size according to its content [CUSTOM]

Fix PCLAS-1233

* Don't overflow select tag arrow with the option text

https://bugs.tryton.org/13765

JCavallo committed:
sao: Theming for sole xxx2many in groups [CUSTOM] (#381)

Issue link: [https://coopengo.atlassian.net/browse/PCLAS-1479](https://coopengo.atlassian.net/browse/PCLAS-1479)

Fix PCLAS-1479

Commit:
Packaging coog-25.03.2503-rc2

Commit:
Packaging coog-25.04.2504-rc2

nicoe committed:
Restore selection widget expansion [CUSTOM] (#382)

Fix PCLAS-1486

nicoe committed:
Prevent forced reflow in tree when setting column value [PREVIEW] (#384)

jQuery's .text() does an .empty() call unconditionally which result in
the browser thinking the DOM has changed even when the textual is
identical

nicoe committed:
Add logging information when a selection value is not found [CUSTOM] (#383)

Fix PCLAS-1488

nicoe committed:
Improve multi-selection behavior and rendering [PREVIEW] (#380)

Fix PCLAS-85

Commit:
Packaging coog-25.05.2505-rc2

JCavallo committed:
sao: Better styling of richtext [CUSTOM] (#386)

Issue link: [https://coopengo.atlassian.net/browse/PCLAS-1508](https://coopengo.atlassian.net/browse/PCLAS-1508)

Fix PCLAS-1508

nicoe committed:
[trytond] Reset the pool instances only on the pool initialization [PREVIEW] (#372)

Fix PCLAS-1280
https://bugs.tryton.org/13763
---------

Co-authored-by: Jean Cavallo <jean.cavallo@hotmail.fr>

JCavallo committed:
sao: Display contextual menu over pop-ups / wizards [CUSTOM] (#387)

Issue link: [https://coopengo.atlassian.net/browse/PCLAS-1512](https://coopengo.atlassian.net/browse/PCLAS-1512)

Fix PCLAS-1512

Commit:
Packaging coog-25.06.2506

Commit:
Packaging coog-25.07.2507-rc3

JCavallo committed:
Sao: Fix z-index of pop-up [CUSTOM]

JCavallo committed:
sao: Set richtext attributes as styling [CUSTOM] (#389)

Issue link: [https://coopengo.atlassian.net/browse/PCLAS-1535](https://coopengo.atlassian.net/browse/PCLAS-1535)

Fix PCLAS-1535

JCavallo committed:
sao: Add missing semicolon [CUSTOM]

Commit:
Packaging coog-25.08.2508-rc2

AntoinePinstonCoop committed:
tryton: Fix file name when exporting url [CUSTOM]

Fix #PROCK-3484

Commit:
Packaging coog-25.09.2509-rc2

Commit:
Packaging coog-25.10.2510-rc2

JCavallo committed:
Merge remote-tracking branch 'origin/master' into PCLAS-1066/jira_col_resizing

JCavallo commented:
Fait quelques tests : 
- La version précédente est plus fluide. Notamment, cette version a un délai (~300ms au feeling) entre le début du drag-n-drop et le moment où le déplacement a lieu visuellement)
- On a le problème (corrigé via un algorighme un peu bancal sur la version précédente) où la colonne qui est resize ne correspond pas à celle dont on a pris le handler

Commit:
Packaging coog-25.11.2511-rc1

nicoe committed:
Fix column width setting and set size of column

nicoe commented:
Sur le délai, j'ai essayé plusieurs techniques mais je l'ai toujours. Je pense que c'est inhérent au fait d'utiliser le drag & drop car dans mes tests, JIRA présente lui aussi ce délai initial.

Sur le problème de la colonne qui n'est pas celle redimensionnée, je ne comprends pas : les colonnes précédentes sont fixées et je ne l'observe pas.

nicoe committed:
Reset completely the overflow for modal-body (#395)

This one is a complicated beast:

    - having defined an `overflow-x: auto` in CSS on modal-body make it
      a new Block Formatting Context
    - Block Formatting Context contains their floating elements
    - This is why although the ul menu is in the same stacking context
      than the modal footer, the menu does not appear above it.
    - Thus setting the `overflow` property will also reset the
      `overflow-x` property and since the value is `visible` it does not
      create a new Block Formatting Context, ensuring the `z-index` take
      effect.

Fix PCLAS-1506

nicoe committed:
Unconditionally enable button after a click [CUSTOM] (#394)

* Unconditionally enable button after a clic [CUSTOM]

PCLAS-1522

* Only call the button state setting when the call to button failed

* Remove console.log

JCavallo committed:
sao: Allow to copy from readonly richtext widget [CUSTOM] (#403)

Issue link: [https://coopengo.atlassian.net/browse/PCLAS-1568](https://coopengo.atlassian.net/browse/PCLAS-1568)

Fix PCLAS-1568

JCavallo committed:
sao: Remove padding in groups with a sole text area (#404)

Issue link: [https://coopengo.atlassian.net/browse/PCLAS-1570](https://coopengo.atlassian.net/browse/PCLAS-1570)

Fix PCLAS-1570

JCavallo committed:
Merge remote-tracking branch 'origin/master' into PCLAS-1066/jira_col_resizing

nicoe committed:
Fix column offset

nicoe committed:
Merge branch 'PCLAS-1066/jira_col_resizing' of github.com:nicoe/tryton-coopengo-new into PCLAS-1066/jira_col_resizing
